clients/src/main/java/org/apache/kafka/clients/admin/RemoveMembersFromConsumerGroupResult.java

```diff
@@ -51,9 +52,21 @@
             if (throwable != null) {
                 result.completeExceptionally(throwable);
             } else {
-                for (MemberToRemove memberToRemove : memberInfos) {
-                    if (maybeCompleteExceptionally(memberErrors, memberToRemove.toMemberIdentity(), result)) {
-                        return;
+                if (removeAll()) {
```

Comment:
Not sure why the `removeAll()` case needs to be handled differently? Can you elaboarte?

Reply:
Because in non `removeAll` scenario, we have put the members to be deleted in the `RemoveMembersFromConsumerGroupResult#memberInfos`, while in the `removeAll` scenario, we don't do so(members to be deleted are decided in the private method: `KafkaAdminClient#getMembersFromGroup` of `KafkaAdminClient`).

Reply:
Well, while `memberInfo` is empty for the `removeAll` case, I am still wondering if the code for `removeAll` would not work for the other case, too?

Reply:
I'm not sure I understand the question, could you elaborate more?

Reply:
Can we just do for both cases?
```
for (Map.Entry<MemberIdentity, Errors> entry: memberErrors.entrySet()) {
    Exception exception = entry.getValue().exception();
    if (exception != null) {
        Throwable ex = new KafkaException("Encounter exception when trying to remove: "
             + entry.getKey(), exception);
        result.completeExceptionally(ex);
        return;
    }
}
```

The "issue" with using `memberInfos` is, that for the removeAll() case it's empty and we cannot use it. However, `memberErrors` should have an entry for all members for both cases?

Reply:
I'm afraid not because, in the non `removeAll` scenario, caller specify the members(`memberInfos`) to be deleted, and according to `maybeCompleteExceptionally`, the `memberInfos` is used because it might sometimes happen that certain member in `memberInfos` cannot be found in `memberErrors `, that's the reason I didn't use the `removeAll` logic for all cases.

Reply:
Thanks for explaining!